src/clangparser.cpp

```diff
@@ -140,6 +140,26 @@ void ClangTUParser::parse()
     {
       argv.push_back(qstrdup(option->c_str()));
     }
+    // The last compile command (last entry of argv) should be the filename of the source
+    // file to parse. It does not matter to clang_parseTranslationUnit below if we pass the file name
+    // separately in its second argument or if we just pass it a nullptr as the second
+    // argument and pass the file name with the other compile commands.
+    // However, in some cases (e.g., starting from Clang 14, if we are parsing a header file, see
+    // https://github.com/doxygen/doxygen/issues/10733), the compile commands returned by
+    // getCompileCommands include a "--" as second to last argument (which is supposed to make it
+    // easier to parse the argument list). If we pass this "--" to clang_parseTranslationUnit below,
+    // it returns an error. To avoid this, we remove the file name argument (and the "--" if present)
+    // from argv and pass the file name separately.
+    if (std::string(argv[argv.size() - 1]) != fileName.str()) {
+      std::cout << fileName.str() << " vs. " << argv[argv.size() - 1] << std::endl;
+    }
```

Comment:
Looks like debug statements, should not be present anymore / commented..

Reply:
Removed.